woodwork/table_accessor.py

```diff
@@ -197,6 +197,9 @@ def __setitem__(self, col_name, column):
         if column.ww.schema is not None and 'index' in column.ww.semantic_tags:
             warnings.warn("Cannot allow adding column with index tags. Tag removed", IndexTagRemovedWarning)
             column.ww.remove_semantic_tags('index')
+            logical_type = column.ww.logical_type
+            if hasattr(logical_type, 'standard_tags'):
```

Comment:
https://github.com/alteryx/woodwork/blob/main/woodwork/logical_types.py#L24-L29

the default if a LogicalType has no specific standard tags is an empty set! So you can just reference `logical_type.standard_tags` without checking that they're present, I think!